databricks_cli/sdk/service.py

```diff
@@ -1050,10 +1051,30 @@ def stop(self, pipeline_id=None, headers=None):
 
         return self.client.perform_query('POST', '/pipelines/{pipeline_id}/stop'.format(pipeline_id=pipeline_id),
                                          data=_data, headers=headers)
+
 class ReposService(object):
+    __git_providers__ = {
+        "github.com":    "gitHub",
+	    "dev.azure.com": "azureDevOpsServices",
+	    "gitlab.com":    "gitLab",
+	    "bitbucket.org": "bitbucketCloud"
+     }
+
     def __init__(self, client):
         self.client = client
 
+    def __detect_repo_provider__(self, url):
+        provider = None
+        try:
+            netloc = urlparse(url).netloc
+            idx = netloc.rfind("@")
+            if idx != -1:
+                netloc = netloc[(idx+1):]
+            provider = ReposService.__git_providers__.get(netloc.lower())
+        except:
+            pass
```

Comment:
do we want to throw some error message here so it's clear to the user they need to specify the git provider if the request fails?

Reply:
Added it in another place instead